aws/data_source_aws_db_cluster_snapshot_test.go

```diff
@@ -91,7 +92,30 @@ func TestAccAWSDbClusterSnapshotDataSource_MostRecent(t *testing.T) {
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccCheckAwsDbClusterSnapshotDataSourceConfig_MostRecent(rName),
+				Config: testAccCheckAwsDbClusterAndSnapshot(rName),
+			},
+			{
+				PreConfig: func() {
+					// To prevent InvalidDBClusterStateFault errors,
+					// ensure rds_cluster is in an "available" state before
+					// creating an additional snapshot resource
+					conn := testAccProvider.Meta().(*AWSClient).rdsconn
+					stateConf := &resource.StateChangeConf{
+						Pending:    resourceAwsRdsClusterCreatePendingStates,
+						Target:     []string{"available"},
+						Refresh:    resourceAwsRDSClusterStateRefreshFunc(conn, rName),
+						Timeout:    15 * time.Minute,
+						MinTimeout: 10 * time.Second,
+						Delay:      30 * time.Second,
+					}
+
+					_, err := stateConf.WaitForState()
+					if err != nil {
+						t.Fatal(err)
+					}
+				},
```

Comment:
In general, if we need to implement custom logic in acceptance testing for other resources, it means we aren't doing something properly in the resource which is also a problem that practitioners won't be able to workaround either (or at least easily). We may want to try adding `ContinuousTargetOccurence: 3` to the `aws_rds_cluster` resource creation retries (which hopefully should catch if RDS goes available -> backing-up -> available) and seeing if the single configuration can apply correctly afterwards. 👍

Reply:
yep totally works! retry time i put is arbitrary though, 2min ok?